src/app/clusters/testing/CommandTesting.h

```diff
@@ -0,0 +1,192 @@
+
+    CHIP_ERROR FallibleAddStatus(const ConcreteCommandPath & aRequestCommandPath,
+                                 const Protocols::InteractionModel::ClusterStatusCode & aStatus,
+                                 const char * context = nullptr) override;
+
+    void AddStatus(const ConcreteCommandPath & aRequestCommandPath, const Protocols::InteractionModel::ClusterStatusCode & aStatus,
+                   const char * context = nullptr) override;
+
+    FabricIndex GetAccessingFabricIndex() const override { return mFabricIndex; }
+
+    CHIP_ERROR AddResponseData(const ConcreteCommandPath & aRequestCommandPath, CommandId aResponseCommandId,
+                               const DataModel::EncodableToTLV & aEncodable) override;
+
+    void AddResponse(const ConcreteCommandPath & aRequestCommandPath, CommandId aResponseCommandId,
+                     const DataModel::EncodableToTLV & aEncodable) override;
+
+    bool IsTimedInvoke() const override { return false; }
+    void FlushAcksRightAwayOnSlowCommand() override {}
+    Access::SubjectDescriptor GetSubjectDescriptor() const override { return Access::SubjectDescriptor{}; }
+    Messaging::ExchangeContext * GetExchangeContext() const override { return nullptr; }
+
+    // Helper methods to extract response data
+    bool HasResponse() const { return !mResponse.encodedData.IsNull(); }
+    CommandId GetResponseCommandId() const { return mResponse.commandId; }
+    const ResponseRecord & GetResponse() const { return mResponse; }
+
+    // Get a TLV reader positioned at the response data fields
+    CHIP_ERROR GetResponseReader(TLV::TLVReader & reader) const;
+
+    // Decode response into a specific DecodableType
+    template <typename ResponseType>
+    CHIP_ERROR DecodeResponse(ResponseType & response) const
+    {
+        TLV::TLVReader reader;
+        ReturnErrorOnFailure(GetResponseReader(reader));
+        return response.Decode(reader);
+    }
+
+    // Configuration methods
+    void SetFabricIndex(FabricIndex index) { mFabricIndex = index; }
+
+private:
+    ResponseRecord mResponse;
+    FabricIndex mFabricIndex = 1; // Default to 1 to maintain backward compatibility
+};
+
+// Helper class for invoking commands in tests.
+// Manages TLV encoding/decoding and command handler setup.
+class InvokeOperation
+{
+public:
+    InvokeOperation(EndpointId endpoint, ClusterId cluster, CommandId command) :
+        mCommandPath(endpoint, cluster, command), mHandler(std::make_unique<MockCommandHandler>())
+    {
+        mRequest.path = mCommandPath;
+    }
+
+    InvokeOperation(const ConcreteCommandPath & path) : mCommandPath(path), mHandler(std::make_unique<MockCommandHandler>())
+    {
+        mRequest.path = mCommandPath;
+    }
+
+    // Invoke a command with no fields (empty TLV structure)
+    // Use for commands that take no parameters
+    template <typename ClusterType>
+    std::optional<DataModel::ActionReturnStatus> InvokeEmpty(ClusterType & cluster)
```

Comment:
```suggestion
    template <typename ClusterType, typename... Args>
std::optional<DataModel::ActionReturnStatus> Invoke(ClusterType & cluster, const Args &... args)
```
Can we combine the two functions in a one, what do you think ?

Reply:
Done